src/serializers/shared.rs

```diff
@@ -28,112 +28,131 @@ pub(crate) trait BuildSerializer: Sized {
     ) -> PyResult<CombinedSerializer>;
 }
 
-/// Build the `CombinedSerializer` enum and implement a `find_serializer` method for it.
-macro_rules! combined_serializer {
-    (
-        enum_only: {$($e_key:ident: $e_serializer:path;)*}
-        find_only: {$($builder:path;)*}
-        both: {$($b_key:ident: $b_serializer:path;)*}
-    ) => {
-        #[derive(Debug, Clone)]
-        #[enum_dispatch]
-        pub enum CombinedSerializer {
-            $($e_key($e_serializer),)*
-            $($b_key($b_serializer),)*
-        }
+#[derive(Debug, Clone)]
```

Comment:
sorry :smile:

I don't agree about removing this macro, it's working, it removes duplication, happy to add a better docstring.

I'd really rather we didn't remove it.